network/common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -56,6 +62,15 @@ public OneForOneStreamManager() {
     // This does not need to be globally unique, only unique to this class.
     nextStreamId = new AtomicLong((long) new Random().nextInt(Integer.MAX_VALUE) * 1000);
     streams = new ConcurrentHashMap<Long, StreamState>();
+    streamIds = new ConcurrentHashMap<Channel, Set<Long>>();
+  }
+
+  @Override
+  public void registerChannel(Channel channel, long streamId) {
```

Comment:
I think we can avoid the new field by doing

```java
streams.get(streamId).associatedChannel = channel;
```

here and

```java
val streamIterator = streams.iterator()
while (streamIterator.hasNext()) {
  StreamState state = streamIterator.next().getValue()
  if (state.associatedChannel == channel) {
    streamIterator.remove();

    // Release all remaining buffers.
    while (state.buffers.hasNext()) {
      state.buffers.next().release();
    }
  }
}
```

Allowing the removal of the other connectionTerminated().